vllm/plugins/__init__.py

```diff
@@ -85,6 +88,13 @@ def set_current_vllm_config(vllm_config: "VllmConfig"):
                      vllm_config.compilation_config.enabled_custom_ops)
         logger.debug("disabled custom ops: %s",
                      vllm_config.compilation_config.disabled_custom_ops)
+        if vllm_config.compilation_config.level == CompilationLevel.PIECEWISE \
+            and compilation_counter.num_models_seen == num_models_seen:
```

Comment:
```suggestion
            and compilation_counter.num_models_seen < num_models_seen:
```

I think you made a typo here. Correct me if I'm wrong though.

Reply:
if the model supports `torch.compile`, we should see `compilation_counter.num_models_seen > num_models_seen`.

`compilation_counter.num_models_seen` is a global counter, and `num_models_seen` is its previous value. `compilation_counter.num_models_seen >= num_models_seen` is always true.

Reply:
Basically we are counting if `TorchCompileWrapperWithCustomDispatcher.__init__` is called. I added more comments. Please check if it makes sense to you.

What is the false positive case?

Reply:
I mean that this check assumes that at least one new model is registered by the plugin. This may not be the case if we consider other types of plugins (perhaps https://github.com/vllm-project/vllm/pull/10546).

Reply:
This is not really a plugin. It is inside the `set_current_vllm_config` function, which is called by `_initialize_model` . The confusing thing might be, `set_current_vllm_config` lives in `vllm/plugins` .

Reply:
Oh, I see now... then this change should be ok. I feel that this function might be more suitable inside `config.py`, but let's not change that in this PR.

Reply:
that's a good idea, I can move it in a followup PR.